### PR TITLE
fix: controls on arcgis map

### PR DESCRIPTION
**Pull request by unclFedor**
also added picking up info

### CODE REVIEW

File: src/hooks/use-arcgis-hook/use-arcgis-hook.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+): unknown | null {
+  const [renderer, setRenderer] = useState<unknown>(null);
+  const [sceneView, setSceneView] = useState<unknown | null>(null);
+  const { longitude, latitude, pitch, bearing, zoom } = {
+    longitude: viewState.main.longitude,
+    latitude: viewState.main.latitude,
+    bearing: viewState.main.bearing,
+    pitch: viewState.main.pitch,
+    zoom: viewState.main.zoom,
+  };
```
Comment (belom88) [inline src/hooks/use-arcgis-hook/use-arcgis-hook.ts:17]:
```suggestion
  const { longitude, latitude, pitch, bearing, zoom } = viewState.main;
```

File: src/hooks/use-arcgis-hook/use-arcgis-hook.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          },
+          zoom: zoom,
+          ui: { components: ["attribution"] },
+          viewingMode: "local",
+        });
+        setSceneView(sceneView);
+
+        sceneView.on("drag", (event) => {
+          if (event.action === "update") {
+            onViewStateChangeHandler({
```
Comment (belom88) [inline src/hooks/use-arcgis-hook/use-arcgis-hook.ts:77]: We should change viewState in Redux here that will reflect onto the map.

File: src/components/map-control-panel/map-control-panel.tsx [hunk 3]
```diff
@@ -144,18 +146,22 @@ export const MapControllPanel = ({
...
+            </Button>
+          )}
+          {isDragModeVisible === true && (
+            <Button
+              active={dragMode === DragMode.rotate}
+              onClick={handleDragModeToggle}
+            >
+              <OrbitIcon />
+            </Button>
+          )}
```
Comment (belom88) [inline src/components/map-control-panel/map-control-panel.tsx:164]:
```suggestion
          {isDragModeVisible === true && (
            <>
            <Button
              active={dragMode === DragMode.pan}
              onClick={handleDragModeToggle}
            >
              <PanIcon />
            </Button>
          )}
            <Button
              active={dragMode === DragMode.rotate}
              onClick={handleDragModeToggle}
            >
              <OrbitIcon />
            </Button>
            </>
          )}
```

File: src/pages/viewer-app/viewer-app.tsx [hunk 4]
```diff
@@ -564,7 +564,27 @@ export const ViewerApp = () => {
           preventTransitions={preventTransitions}
         />
       )}
-      {selectedBaseMapId === "ArcGis" && <ArcgisWrapper />}
+      {selectedBaseMapId === "ArcGis" && (
+        <ArcgisWrapper
```
Comment (belom88) [inline src/pages/viewer-app/viewer-app.tsx:568]: It seems `DeckGlWrapper` and `ArcgisWrapper` have same input. You can

```
const MapWrapper = selectedBaseMapId === "ArcGis" ? ArcgisWrapper : DeckGlWrapper;
...
return ......
  <MapWrapper ...
```

**Review by belom88 — CHANGES_REQUESTED** [review]
Create a file in utils to put code shared between `ArcGisWrapper` and `DeckGlWrapper`. For hooks we can create a custom hook to share in those components.

File: src/pages/viewer-app/viewer-app.tsx (hunk 4 above)
Comment (unclFedor) [reply src/pages/viewer-app/viewer-app.tsx:568]: 1. Aligned input parameters between DeckGlWrapper and ArcgisWrapper.
2.  Added recommended construction:
const MapWrapper = selectedBaseMapId === "ArcGis" ? ArcgisWrapper : DeckGlWrapper;
...
return ......
  <MapWrapper ...

File: src/components/map-control-panel/map-control-panel.tsx (hunk 3 above)
Comment (unclFedor) [reply src/components/map-control-panel/map-control-panel.tsx:164]: move buttons under single condition as recommended

File: src/hooks/use-arcgis-hook/use-arcgis-hook.ts (hunk 1 above)
Comment (unclFedor) [reply src/hooks/use-arcgis-hook/use-arcgis-hook.ts:17]: Updated code per suggestion.

**Review by belom88 — APPROVED** [review]

**Pull request merged**